src/App.js

```diff
@@ -0,0 +1,160 @@
+    }
+  }
+  handlePeopleChange(e) {
+    const numberOfPeople = e.target.value;
+    if (numberOfPeople === "" || /^\d+$/.test(numberOfPeople)) {
+      this.setState({ numberOfPeople });
+    }
+  }
+  handleFirstPayPercentChange(e) {
+    const firstPayPercent = e.target.value;
+    if (firstPayPercent === "" || /^\d{0,2}$/.test(firstPayPercent)) {
+      const firstPayAmount = roundTo(
+        firstPayPercent * 0.01 * this.state.billAmount,
+        2
+      );
+      this.setState({ firstPayPercent, firstPayAmount });
+    }
+  }
+  render() {
+    const calculateFinalAmount = () => {
+      if (!this.state.advancedVersion) {
+        return (
+          "$" + roundTo(this.state.billAmount / this.state.numberOfPeople, 2)
+        );
+      }
+      return (
+        "$" +
+        roundTo(
+          (this.state.billAmount - this.state.firstPayAmount) /
+            (this.state.numberOfPeople - 1),
+          2
+        )
+      );
+    };
+    return (
+      <div>
+        <Container>
+          <Row>
+            <Col xs="0" sm="3" md="3"></Col>
+            <Col xs="12" sm="6">
+              <h1>Bill Splitter</h1>
+              <h4>created by Andrew Horn</h4>
+              <br />
+              <br />
+              <Button color="primary" onClick={this.toggleAdvanced}>
+                {this.state.advancedVersion
+                  ? "Advanced Version"
+                  : "Basic Version"}
+              </Button>
+              <br />
+              <br />
+              <Form>
+                <Label>Bill Amount</Label>
+                <InputGroup>
+                  <InputGroupAddon addonType="prepend">$</InputGroupAddon>
+                  <Input
+                    type="text"
+                    value={this.state.billAmount}
+                    onChange={this.handleBillChange}
+                  />
+                </InputGroup>
+                <br />
+                <Label>Number of people</Label>
+                <Input
+                  type="text"
+                  value={this.state.numberOfPeople}
+                  onChange={this.handlePeopleChange}
+                />
+                <br />
+                {this.state.advancedVersion && (
+                  <div>
+                    <Label>Percent 1st person pays</Label>
+                    <InputGroup>
+                      <Input
+                        type="text"
+                        value={this.state.firstPayPercent}
+                        onChange={this.handleFirstPayPercentChange}
+                      />
+                      <InputGroupAddon addonType="append">%</InputGroupAddon>
+                    </InputGroup>
+                  </div>
+                )}
+                <br />
+                <br />
+                <br />
+                {this.state.billAmount &&
+                  this.state.numberOfPeople &&
+                  (!this.state.advancedVersion ||
+                    this.state.firstPayPercent) && (
```

Comment:
this complex piece of conditional logic should be done in a separate function if possible